pytorch_lightning/loggers/base.py

```diff
@@ -27,20 +31,62 @@ class LightningLoggerBase(ABC):
 
     def __init__(self):
         self._rank = 0
+        self._prev_step = -1
+        self._metrics_to_agg: List[Counter[str, float]] = []
 
     @property
     @abstractmethod
     def experiment(self) -> Any:
         """Return the experiment object associated with this logger"""
 
+    def _agg_metrics(self, metrics: MetricsT, step: Optional[int] = None) -> Optional[MetricsT]:
+        """Aggregates metrics.
+
+        Args:
+             Check `log_metrics` method for the arguments description.
+
+        Returns (MetricsT, None):
+            Aggregated metrics. The return value could be None. In such case, metrics
+            are added to the aggregation list, but not aggregated yet.
+        """
+        if step == self._prev_step:
+            self._metrics_to_agg.append(collections.Counter(metrics))
+            return None
+
+        if len(self._metrics_to_agg) > 0:
+            agg_mets = functools.reduce(lambda c1, c2: c1 + c2, self._metrics_to_agg)
+            agg_mets = {k: v / len(self._metrics_to_agg) for k, v in agg_mets.items()}
```

Comment:
should we factor out the aggregation function as an op (similar to [here[(https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/callbacks/early_stopping.py#L59))? and perhaps the reduction op is passed into the logger at initialization? i think a `mean` reduction will be most common, but it might be nice to be able to do `min` and `max` reductions as well.

Reply:
good idea! done!